File: frontend/lib/data/models/route_schedule.dart
class RouteSchedule {
  final String dayType;
  final String firstTime;
  final String lastTime;
  final String scheduleType;

  const RouteSchedule({
    required this.dayType,
    required this.firstTime,
    required this.lastTime,
    required this.scheduleType,
  });

  factory RouteSchedule.fromJson(Map<String, dynamic> json) => RouteSchedule(
    dayType: json['DayType'] as String? ?? '',
    firstTime: json['FirstTime'] as String? ?? '',
    lastTime: json['LastTime'] as String? ?? '',
    scheduleType: json['ScheduleType'] as String? ?? '',
  );
}

File: backend/src/main/java/com/nusmotion/backend/dto/RouteSchedule.java
package com.nusmotion.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Operating schedule row from /RouteMinMaxTime.
 */
public record RouteSchedule(
        @JsonProperty("DayType") String dayType,
        @JsonProperty("FirstTime") String firstTime,
        @JsonProperty("LastTime") String lastTime,
        @JsonProperty("ScheduleType") String scheduleType
) {}